device/iot-device-samples/device-reconnection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/Application.java

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft. All rights reserved.Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+package samples.com.microsoft.azure.sdk.iot;
+
+import com.microsoft.azure.sdk.iot.device.DeviceClient;
+import com.microsoft.azure.sdk.iot.device.DeviceTwin.Pair;
+import com.microsoft.azure.sdk.iot.device.DeviceTwin.Property;
+import com.microsoft.azure.sdk.iot.device.DeviceTwin.TwinPropertyCallBack;
+import com.microsoft.azure.sdk.iot.device.IotHubClientProtocol;
+import com.microsoft.azure.sdk.iot.device.Message;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static samples.com.microsoft.azure.sdk.iot.DeviceTwinHelper.*;
+import static samples.com.microsoft.azure.sdk.iot.MethodHelper.DeviceMethodStatusCallBack;
+import static samples.com.microsoft.azure.sdk.iot.MethodHelper.SampleDeviceMethodCallback;
+import static samples.com.microsoft.azure.sdk.iot.ReceiveMessageHelper.MessageCallback;
+import static samples.com.microsoft.azure.sdk.iot.TelemetryHelper.EventCallback;
+import static samples.com.microsoft.azure.sdk.iot.TelemetryHelper.composeMessage;
+
+@Slf4j
+public class Application {
+    private static final String deviceConnectionString = System.getenv("DEVICE_CONNECTION_STRING");
+    private static final IotHubClientProtocol protocol = IotHubClientProtocol.AMQPS; // Can be configured to use any protocol from  HTTPS, AMQPS, MQTT, AMQPS_WS, MQTT_WS
```

Comment:
>HTTPS [](start = 132, length = 5)

HTTPS doesn't support connection status callback, so this is a little misleading

Reply:
I'll add a note stating that status callback, twins and methods are not supported on HTTP